atomic_reactor/util.py

```diff
@@ -708,6 +725,16 @@ def get_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
             insecure=insecure, dockercfg_path=dockercfg_path,
             version=version)
 
+        # If the registry has a v2 manifest that can't be converted into a v1
+        # manifest, the registry fails with status=400, and a error code of
+        # MANIFEST_INVALID.
+        if response.status_code == 400:
```

Comment:
The http standard gives two options when the Accept header cannot be honored: return '406 - Not Acceptable' or ignore the Accept header. So considering that, checking for v1 isn't unreasonable, and might avoid hiding actual client errors (400 is supposed to be when the request is in error.)

The question of what status codes can be returned in similar cases is complicated:
 * If conversion fails because data on the server has invalid references (that should have been caught at upload), then 400 is also returned. 500 can be returned in other cases of conversion failure.
 * If there is a manifest list on the server, but that doesn't contain any manifest for the default arch+os, then 404 is returned.
 * In the draft patch for OCI support, then requesting a docker manifest when only an OCI manifest is available or vice-versa also returns a 404.

The most conservative thing in some ways would to be to treat a 404 or a 406 as a not-log-and-continue, and to treat any other error status as a log-and-continue.